### PR TITLE
[CARBONDATA-3208] Remove unused parameters, imports and optimize the spell errors

### PR DESCRIPTION
**Pull request by runzhliu**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 
 - [x] Any backward compatibility impacted?
 
 - [x] Document update required?

 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by qiuchenjian** [comment]
@xuchuanyin   @xubo245 @QiangCai  @jackylk  Can we enable checkstyle rule of avoiding unused imports，i think it's necessary to keep good code style and reduce the dependency of a class on the number of jar packages,  the rule may be is   <module name="UnusedImports"/>

**Comment by xubo245** [comment]
@qiuchenjian I tried it before, but there are some limit, not easy to support it. If you have better way to support it, you can raise a PR to implement it.

**Comment by xubo245** [comment]
add to whitelist

**Comment by qiuchenjian** [comment]
@xubo245 OK, i'll try

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/PartitionFactory.scala [hunk 1]
```diff
@@ -32,7 +32,7 @@ object PartitionFactory {
       case PartitionType.LIST => new ListPartitioner(partitionInfo)
       case PartitionType.RANGE => new RangePartitioner(partitionInfo)
       case partitionType =>
-        throw new CarbonDataLoadingException(s"Unsupport partition type: $partitionType")
+        throw new CarbonDataLoadingException(s"Unsupported partition type: $partitionType")
```
Comment (xubo245) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/PartitionFactory.scala]: Can you check it in whole project, there are another two.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2075/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala [hunk 2]
```diff
@@ -483,12 +483,12 @@ class CarbonGlobalDictionaryGenerateRDD(
 }
 
 /**
- * Set column dictionry patition format
```
Comment (xubo245) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala]: Have you check it in the whole project? patition has 25+

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 3]
```diff
@@ -387,7 +382,7 @@ object GlobalDictionaryUtil {
    * @param table         carbon table identifier
    * @param colName       user specified  column name for predefined dict
    * @param colDictPath   column dictionary file path
-   * @param parentDimName parent dimenion for complex type
+   * @param parentDimName parent dimension for complex type
```
Comment (xubo245) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: Have you check it in the whole project?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10329/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/PartitionFactory.scala (hunk 1 above)
Comment (runzhliu) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/PartitionFactory.scala]: Sure, I got another three.

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala (hunk 2 above)
Comment (runzhliu) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala]: OK

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2280/

**Comment by chenliang613** [comment]
@runzhliu  please correct the PR title.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2076/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2281/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10330/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2078/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2283/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10332/

**Comment by xubo245** [comment]
Please optimize the title， this PR is not only for unused import, also optimize the spell error, so please add it too. @runzhliu

**Comment by runzhliu** [comment]
@xubo245 Done.

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2087/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2292/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10341/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2089/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2294/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10343/

**Comment by chenliang613** [comment]
LGTM, thanks for the good contributions.

File: core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentProperties.java [hunk 4]
```diff
@@ -287,31 +287,31 @@ private void fillDimensionAndMeasureDetails(List<ColumnSchema> columnsInTable,
...
           carbonDimension =
-              new CarbonDimension(columnSchema, dimensonOrdinal++, -1, ++complexTypeOrdinal);
+              new CarbonDimension(columnSchema, dimensionOrdinal++, -1, ++complexTypeOrdinal);
           carbonDimension.initializeChildDimensionsList(columnSchema.getNumberOfChild());
           complexDimensions.add(carbonDimension);
           isComplexDimensionStarted = true;
-          int previouseOrdinal = dimensonOrdinal;
-          dimensonOrdinal =
-              readAllComplexTypeChildren(dimensonOrdinal, columnSchema.getNumberOfChild(),
+          int previouseOrdinal = dimensionOrdinal;
```
Comment (xuchuanyin) [inline core/src/main/java/org/apache/carbondata/core/datastore/block/SegmentProperties.java]: um, part of the variable name 'previouse' is also typo, which should be 'previous'...
Please correct this also and others LGTM.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2092/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2297/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10346/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2093/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2298/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10347/

**Comment by xubo245** [comment]
LGTM. Thank you for working on it.

**Pull request closed without merging**